### PR TITLE
Add a home screen, clear missing projects option, new icons, close project, preferences refresh

### PR DESCRIPTION
**Pull request by WiIIiam278**
This pull request does a bunch of UI work
* New home screen, containing recent projects and good options to get started with
* The ability to close a project and return to this screen
* An update to the preferences screen, that'll make it easier to add new options to it (and it's now a little more organised)
* Several new icons, including for preferences, project settings, saving, etc

<details>
<summary>Home screen</summary>

image
</details>
<details>
<summary>Preferences refresh</summary>

image</details>
<details>
<summary>New icons, close projects</summary>

image
</details>

### CODE REVIEW

File: src/SerialLoops/Dialogs/PreferencesDialog.cs [hunk 1]
```diff
@@ -1,9 +1,15 @@
 using Eto.Drawing;
 using Eto.Forms;
 using HaruhiChokuretsuLib.Util;
+using Microsoft.VisualBasic;
```
Comment (jonko0493) [inline src/SerialLoops/Dialogs/PreferencesDialog.cs:4]: I feel like there's no way we're actually using this, right?

File: src/SerialLoops/Dialogs/PreferencesDialog.cs [hunk 2]
```diff
@@ -1,9 +1,15 @@
...
 using SerialLoops.Lib;
 using SerialLoops.Utility;
 using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Xml.Linq;
```
Comment (jonko0493) [inline src/SerialLoops/Dialogs/PreferencesDialog.cs:11]: I don't think we're doing anything with XML, right?

File: src/SerialLoops/Dialogs/PreferencesDialog.cs [hunk 3]
```diff
@@ -136,42 +111,115 @@ private void SaveButton_Click(object sender, EventArgs e)
             Close();
         }
 
-        private void DevkitArmBox_TextChanged(object sender, EventArgs e)
+        internal class OptionsGroup : GroupBox
```
Comment (jonko0493) [inline src/SerialLoops/Dialogs/PreferencesDialog.cs:114]: Really love this options setup -- great work!

File: src/SerialLoops/MainForm.eto.cs [hunk 4]
```diff
@@ -22,83 +22,132 @@ public partial class MainForm : Form
...
+            {
+                ToolBar = null;
+            }
 
-            Command buildBaseProject = new() { MenuText = "Build from Scratch", ToolBarText = "Build from Scratch", Image = ControlGenerator.GetIcon("Build_Scratch", _log) };
-            buildBaseProject.Executed += BuildBaseProject_Executed;
-            
-            Command buildAndRunProject = new() { MenuText = "Build and Run", ToolBarText = "Run", Image = ControlGenerator.GetIcon("Build_Run", _log) };
-            buildAndRunProject.Executed += BuildAndRunProject_Executed;
+            GC.Collect();
```
[the remark this answers is not in the document]
Comment (WiIIiam278) [reply src/SerialLoops/MainForm.eto.cs:78]: Not particularly, but a lot is getting unloaded at once here. Gonna remove it though, you're right that it's probably unneccessary.

**Pull request merged**